src/components/Droppable.tsx

```diff
@@ -0,0 +1,19 @@
+import { useDrop } from "react-dnd";
+
+interface DroppableProps {
+  type: string;
+  index: number;
+  children: React.ReactElement;
+}
+
+export function Droppable({ type, index, children }: DroppableProps) {
+  const [, drop] = useDrop(
+    () => ({
+      accept: type,
+      drop: () => ({ index }),
```

Comment:
So, Droppable component shouldn't care about anything that's why its only concern is to pass the index of the change. There is another component called Draggable that has the onMove prop (it also doesn't know anything about the logic only when to trigger it). Finally, we have the moveTalkingpoint function that deals with it and there the strategy can vary as much as we want, Optimistic UI, Pessimistic UI doesn't really matter because this chances are isolated there